src/validation.cpp

```diff
@@ -4493,7 +4493,7 @@ void CChainState::CheckBlockIndex(const Consensus::Params& consensusParams)
 
     // Build forward-pointing map of the entire block tree.
     std::multimap<CBlockIndex*,CBlockIndex*> forward;
-    for (auto& entry : mapBlockIndex) {
+    for (const auto& entry : mapBlockIndex) {
```

Comment:
Most common usage when iterating `mapBlockIndex` is
```cpp
for (const std::pair<const uint256, CBlockIndex*>& item : mapBlockIndex)
```
And there is also the above:
```cpp
for (const BlockMap::value_type& entry : mapBlockIndex)
```
(I prefer the 1st).